### ISSUE TITLE
plugin check base os version

### ISSUE THREAD

**Issue by limetech** [limetech/dynamix#65]
Something to ponder: there needs to be a way for a plugin file to specify the minimum "base os" version the plugin is compatible with.

For example, say we release unRaid base os version 6.0-beta14 which includes dynamix webGui code that's tagged maybe "2015-02-20".  Once someone installs -beta14, when they go to plugin page and check for updates, it will show no updates available (correct).

But if someone is running say -beta12, when they go to plugin page and check for updates, it will say there is both a dynamix update available and an unRaid Server OS update available.

The problem is that if the dynamix update should not be shown in this case because it's "incompatible" with the -beta12 release they are running.

Need to fix this...

**Comment by bergware** [comment]
Agreed that it should not show the update as available, but it won't install anyway in this case.

Perhaps a new optional attribute can be introduced in the PLG file, say "baseOS". When present it designates the lowest required OS version, if omitted it is assumed the plugin runs on any OS version.

The "plugins.page" script can take this attribute into account and display updates accordingly. One challenge though, if we move like v6.0beta13, v6.0beta14 and then v6.0, then how is the latter one recognized as the successor of the beta versions?

**Comment by eschultz** [comment]
If we use the guidelines from here: http://semver.org/  We should be able to get our pre-release (beta, rc) versions to work correctly when comparing the min OS version.

I've always been a fan of how Node/NPM use semver to deal with dependencies.  Instead of minversion they have a bit more flexibility, see this neat simulator on how matching and operators work: http://semver.npmjs.com/

**Comment by eschultz** [comment]
I just stumbled on to this php function the other day: [version_compare](http://php.net/manual/en/function.version-compare.php)

Looks helpful and handles the 'beta', 'alpha', 'rc', etc suffixes intelligently.

**Comment by bergware** [comment]
Talking about PHP versions, PHP-5.4 is not maintained any longer and a good successor will be PHP-7. Any consideration?

**Comment by eschultz** [comment]
unRAID 6.2 will ship with PHP-5.6.  I looked in to using PHP-7 but libvirt-php (php extension) would have to be ported first since there's a bunch of underlying changes in PHP-7.
